lib/puppet/indirector/catalog/compiler.rb

```diff
@@ -267,7 +267,7 @@ def compile(node, options)
           Puppet.err(detail.to_s) if networked?
           raise
         ensure
-          Puppet::Type.clear_misses if Puppet[:always_cache_misses]
+          Puppet::Type.clear_misses if Puppet[:always_retry_plugins]
```

Comment:
no, it needs a not or unless. i missed that in the reversal. this is here so that if a user accidentally uses a type in a manifest before it is available, that the type lookup failure is only cached for the duration of the compile. that maintains backward compatibility with current behavior. otherwise if the user used a type before it was available, they would need to flush the jruby pool or restart the server.